pyntc/devices/ios_device.py

```diff
@@ -69,6 +69,18 @@ def _raw_version_data(self):
             return version_data
         except IndexError:
             return {}
+            
+    def _reconnect(self, timeout=60):
+        counter = 0
+        timeout = timeout*4
+        while counter < timeout:
+            try:
+                self.open()
+                return True
+            except:
+                counter += 1
+                time.sleep(15)
+        raise ValueError('reconnect timeout: could not verified device upgrade')
```

Comment:
The exception should be generic since it is not necessarily tied to upgrading the OS.